specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json

```diff
@@ -0,0 +1,286 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationIdentityClient",
+    "description": "Azure Communication Identity Service",
+    "version": "2021-03-07"
+  },
+  "paths": {
+    "/identities": {
+      "post": {
+        "tags": [
+          "Identity"
+        ],
+        "summary": "Create a new identity.",
+        "operationId": "CommunicationIdentity_CreateIdentity",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "api-version",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityCreateRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the created identity.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessTokenResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create an Identity": {
+            "$ref": "./examples/CreateIdentity.json"
+          }
+        }
+      }
+    },
+    "/identities/{id}": {
+      "delete": {
+        "tags": [
+          "Identity"
+        ],
+        "summary": "Delete the identity, revoke all tokens for the identity and delete all associated data.",
+        "operationId": "CommunicationIdentity_DeleteIdentity",
```

Comment:
The "Identity" suffix on the method is redundant.

Reply:
operationId is consistent to SDK needs.  
![image](https://user-images.githubusercontent.com/17181592/104748445-0bb0c100-5752-11eb-974a-961f03d78528.png)

Reply:
The convention is [`NOUN_VERB`](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r1001-operationidnounverb), not the other way around.

There are numerous examples of this pattern, I will cite [`redis`](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/redis/resource-manager/Microsoft.Cache/stable/2020-06-01/redis.json#L173) as the canonical example because their swaggers are simple and high-quality.

With the current name, e.g. in C# you'll generate a class named `CommunicationIdentity` with a method `CreateIdentity`.  Is this really what you want?

Reply:
Sure not, let me ask SDK , because I was told  OperationId "CommunicationIdentity_Create" will generate CommunicationIdentityRestClient.Create .

Reply:
Updated